app/controllers/api/v0/service_plans_controller.rb

```diff
@@ -4,6 +4,12 @@ class ServicePlansController < ApplicationController
       include Api::Mixins::IndexMixin
       include Api::Mixins::ShowMixin
 
+      def order
+        service_plan = model.find(params[:id])
+        task_id = service_plan.order(params[:catalog_id], params[:other_params?])
```

Comment:
@eclarizio When a plan is ordered there are 2 sets of JSON parameter blobs

1. That is provider specific e.g. {'namespace' => 'my_project', 'OpenShift_param1' => 'xxxx'}
2. Plan Specific e.g. {'DB_NAME' => 'TEST_DB', 'namespace' => 'MARIA_DB_NAMESPACE1'}

We are collecting these 2 separately and sending it to the topology service

https://github.com/ManageIQ/service_portal-api/blob/db46ebe37150ac9c4f896fade228e0e2c8d91641/public/doc/swagger-2.yaml#L523

https://github.com/ManageIQ/service_portal-api/blob/db46ebe37150ac9c4f896fade228e0e2c8d91641/public/doc/swagger-2.yaml#L526

Reply:
@mkanoor is the provider specific json param a part of the service plan json spec? Should it have been??